greymatter-io/operator
# WIP: greymatter.io/network-mode label

This pr adds the ability to add the annotation `greymatter.io/exclude-from-mesh: "true"` to deployments and stateful sets and no sidecar will be added and mesh configs wont be created.

To test:
- deploy the operator 
- create a mesh
- deploy a deployment or stateful set into a watched namespace with the above annotation
- the resource should be created and deleted as desired.  Check to ensure no sidecar was appended to the pod.

Im sure there is a way to abstract the deployment/stateful set switch block but im unsure how to do that.  I think we need to abstract this to a client.Object.  Would love to pair program with someone if anyone wants to refactor this.
